_posts/2023-06-20-introducing-iam-for-openfaas.md

```diff
@@ -0,0 +1,479 @@
+```
+
+Create a secret for your OpenFaaS license:
+
+```bash
+kubectl create secret generic \
+  -n openfaas \
+  openfaas-license \
+  --from-file license=$HOME/.openfaas/LICENSE
+```
+
+Add the OpenFaaS helm chart repo:
+
+```
+helm repo add openfaas https://openfaas.github.io/faas-netes/
+```
+
+Configure ingress to make the OpenFaaS gateway accessible to the users.
+
+The below instructions show how to set up Ingress with a TLS certificate using Ingress Nginx. You can also use any other ingress-controller, inlets-pro or an Istio Gateway.
+
+Install [cert-manager](https://cert-manager.io/docs/), which is used to manage TLS certificates.
+
+You can use Helm, or [arkade](https://github.com/alexellis/arkade):
+
+```bash
+arkade install cert-manager
+```
+
+Install ingress-nginx using arkade or Helm:
+
+```bash
+arkade install ingress-nginx
+```
+
+Create an ACME certificate issuer:
+
+```bash
+export EMAIL="mail@example.com"
+
+cat > issuer-prod.yaml <<EOF
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: letsencrypt
+spec:
+  acme:
+    server: https://acme-v02.api.letsencrypt.org/directory
+    email: $EMAIL
+    privateKeySecretRef:
+      name: letsencrypt
+    solvers:
+    - http01:
+        ingress:
+          class: nginx
+EOF
+```
+
+```bash
+kubectl apply -f issuer-prod.yaml
+```
+
+Create an ingress record for the gateway:
+
+```bash
+export DOMAIN="gateway.openfaas.example.com"
+
+cat > gateway-ingress.yaml <<EOF
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  annotations:
+    cert-manager.io/cluster-issuer: letsencrypt
+    kubernetes.io/ingress.class: nginx
+  labels:
+    app: gateway
+  name: gateway
+  namespace: openfaas
+spec:
+  rules:
+  - host: $DOMAIN
+    http:
+      paths:
+      - backend:
+          service:
+            name: gateway
+            port:
+              number: 8080
+        path: /
+        pathType: Prefix
+  tls:
+  - hosts:
+    - $DOMAIN
+    secretName: gateway-cert
+EOF
+```
+
+```bash
+kubectl apply -f gateway-ingress.yaml
+```
+
+Create a signing key for the OpenFaaS issuer. It is used by the OIDC plugin to sign access tokens issued by OpenFaaS.
+
+```bash
+# Generate a key
+openssl ecparam -genkey -name prime256v1 -noout -out issuer.key
+
+# Store in a secret in the openfaas namespace
+kubectl -n openfaas \
+  create secret generic issuer-key \
+  --from-file=issuer.key=./issuer.key
+```
+
+Create a values.yaml file with your OpenFaaS configuration:
+
+```yaml
+openfaasPro: true
+
+operator:
+  create: true
+
+clusterRole: true
+
+gateway:
+  replicas: 3
+
+queueWorker:
+  replicas: 3
+
+queueWorkerPro:
+  maxInflight: 50
+
+queueMode: jetstream
+
+iam:
+  enabled: true
+  # The system issuer url is the public url the gateway is accessible at.
+  systemIssuer:
+    url: https://gateway.openfaas.example.com
+```
+
+In addition to the configuration for IAM this values.yaml file contains the recommended parameters for running OpenFaaS Pro.
```

Comment:
values-iam.yaml everywhere please

Not to conflict with what the chart ships

Reply:
Renamed this.